kernel/common/hash.rb

```diff
@@ -632,6 +632,9 @@ def [](key)
     end
   end
 
+  # Used internally to get around subclasses redefining #[]
+  alias_method :__get__, :[]
+
```

Comment:
It's better to either use a mirror (and have `#[]` also use that), or simply make the statement that if people overwrite `#[]` in a subclass its their responsibility to also ensure `#to_proc` works. I'd prefer the latter since the former is an endless game of whack-a-mole we can't win.